packages/gestalt/src/FormHelperText.js

```diff
@@ -1,21 +1,92 @@
 // @flow strict
-
-import { type Node } from 'react';
+import { Fragment, useRef, useEffect, useState, type Node } from 'react';
 import Box from './Box.js';
+import Flex from './Flex.js';
+import Status from './Status.js';
 import Text from './Text.js';
+import { type MaxLength } from './TextField.js';
 
 type Props = {|
-  text: string,
+  text: ?string,
+  maxLength?: ?MaxLength,
+  currentLength?: number,
   addA11yPause?: boolean,
 |};
 
-export default function FormHelperText({ text, addA11yPause = false }: Props): Node {
+export default function FormHelperText({
+  currentLength,
+  text,
+  maxLength,
+  addA11yPause = false,
+}: Props): Node {
+  const ref = useRef();
+  const [width, setWidth] = useState(undefined);
+
+  useEffect(() => {
```

Comment:
Why is this logic needed? This is a lot of additional complexity — can this be solved with Flex instead?

Reply:
This is calculating the max with this counter can have. We need to know the width of 100/100 so the width of the counter can be set and keep static